### PR TITLE
added guards against invalid requeset and product statuses

### PR DESCRIPTION
**Pull request by nanstc**
This PR guards against invalid request AND product statuses (didn't foresee the second one coming) in RequestController.java. 

Specifically a template method was created to iterate through enum classes and compare input with defined enums. If strings do not match an new InvalidStatusCodeException is thrown (in the case of product status mismatch) and controller returns 400 (bad request).

### CODE REVIEW

**Comment by csemiao** [comment]
There's an unused guard in requests controller for the put method (it's commented out at the bottom).  While you're looking at guards do you think you can test and confirm it?

File: src/main/java/com/coastcapitalsavings/mvc/controllers/RequestsController.java [hunk 1]
```diff
@@ -237,23 +243,42 @@
...
+        boolean contained = false;
+        for (E status: e.getEnumConstants()) {
+            if (statusCode.equals(status.name())){
+                contained = true;
+                break;
+            }
+        }
+        return contained;
+    }
 }
```
Comment (csemiao) [inline src/main/java/com/coastcapitalsavings/mvc/controllers/RequestsController.java]: Neat!

**Comment by nanstc** [comment]
> There's an unused guard in requests controller for the put method (it's commented out at the bottom). While you're looking at guards do you think you can test and confirm it?

@csemiao I tried to look for it in RequesetController but can't find any commented out guard o.O which line(s)?

**Comment by csemiao** [comment]
sorry, it's in request service.  right at the bottom.

**Comment by nanstc** [comment]
Gotcha. Testing now.

**Pull request merged**